OpenSim/Analyses/JointReaction.cpp

```diff
@@ -284,78 +279,52 @@ void JointReaction::setupReactionList()
-        int validJointFlag = 0;
-        for (int j=0; j<numJoints; j++) {
-            Joint& joint = jointSet.get(j);
-            if (_jointNames.get(i) == joint.getName()) {
-                validJointFlag++;
-                listNotEmptyFlag++;
-                currentJoint.jointName = joint.getName();
-                const std::string& childName = joint.getChildBodyName();
-                int childIndex = bodySet.getIndex(childName, 0);
-                const std::string& parentName = joint.getParentBodyName();
-                int parentIndex = bodySet.getIndex(parentName, 0);
-
-                /* set index that correponds to the appropriate index of the 
-                *  computeReactions arguements forcesVec and momentsVec.*/
-                currentJoint.reactionIndex = childIndex;
-
-                /* set the onBodyIndex to either the parent or child body*/
-                std::string whichBody ="child";
-                if (_onBody.getSize() == 1) {
-                    whichBody = _onBody[0];
-                    }
-                else whichBody = _onBody[i];
-                
-                //convert whichBody to lower case
-                std::transform(whichBody.begin(),whichBody.end(),whichBody.begin(),::tolower);
-
-                if (whichBody == "parent") {
-                    currentJoint.onBodyIndex = parentIndex;}
-                else if(whichBody == "child") {currentJoint.onBodyIndex = childIndex;}
-                else {
-                    currentJoint.onBodyIndex = childIndex;
-                    cout << "\nWARNING:  " << whichBody << " is not a valid choice for apply_on_body";
-                    cout << "\nSetting to apply " << currentJoint.jointName << " load to the child body.\n";}
-
-                /* set the inFrameIndex to either the ground, child, or parent*/
-                std::string whichFrame = "ground";
-                if (_inFrame.getSize() == 1) {
-                    whichFrame = _inFrame[0];}
-                else whichFrame = _inFrame[i];
-
-                // convert to lower case
-                std::transform(whichFrame.begin(),whichFrame.end(),whichFrame.begin(),::tolower);
-
-
-                if (whichFrame == "child") {
-                    currentJoint.inFrameIndex = childIndex;}
-                else if (whichFrame == "parent") {
-                    currentJoint.inFrameIndex = parentIndex;}
-                else if (whichFrame == "ground") {currentJoint.inFrameIndex = groundIndex;}
-                else {
-                    currentJoint.inFrameIndex = groundIndex;
-                    cout << "\nWARNING:  " << whichFrame << " is not a valid choice for express_in_frame";
-                    cout << "\nSetting to express " << currentJoint.jointName << " load in the ground frame.\n";}
-
-                _reactionList.append(currentJoint);
-                break;
+    int index = -1;
+    for (int i = 0; i < _jointNames.getSize(); i++) {
```

Comment:
[CONTRIBUTING.md](https://github.com/opensim-org/opensim-core/blob/master/CONTRIBUTING.md#always-use-pre-increment-and-pre-decrement-operators-when-you-have-a-choice) demands that we use pre-increment wherever possible.